xDrip/Extensions/TimeFrameSelectionView.swift

```diff
@@ -0,0 +1,231 @@
+    
+    typealias SegmentChangeCallback = (Int) -> ()
+    
+    private let backgroundViewCornerRadius: CGFloat = 9.0
+    private let selectedSegmentViewCornerRadius: CGFloat = 7.0
+    
+    private let separatorWidthConstraintConstant: CGFloat = 1.0
+    private let separatorHeightConstraintConstant: CGFloat = 12.0
+    
+    private let labelRegularFont = UIFont.systemFont(ofSize: 16.0, weight: .regular)
+    private let labelMediumFont = UIFont.systemFont(ofSize: 16.0, weight: .medium)
+    
+    private var selectedSegmentBackgroundView = UIView()
+    private var selectedSegmentViewWidthConstraint: NSLayoutConstraint? = nil
+    private var selectedSegmentViewLeadingConstraint: NSLayoutConstraint? = nil
+    
+    private var separators: [UIView] = []
+    private var stackView = UIStackView()
+    private var selectedSegment = 0
+    
+    private var hiddenSeparators: [UIView] = []
+    
+    private var singleSegmentWidth: CGFloat = 0.0
+    
+    private var buttons: [String] = []
+    
+    var segmentChangedHandler: SegmentChangeCallback?
+    
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        
+        setupUI()
+    }
+    
+    override var bounds: CGRect {
+        didSet {
+            self.singleSegmentWidth = (self.frame.width - CGFloat(4 * buttons.count)) / CGFloat(buttons.count)
+            self.selectedSegmentViewWidthConstraint?.constant = singleSegmentWidth
+            
+            self.setupSeparators()
+            self.bringSubviewToFront(selectedSegmentBackgroundView)
+            self.bringSubviewToFront(stackView)
+        }
+    }
+    
+    private func setupUI() {
+        self.layer.cornerRadius = backgroundViewCornerRadius
+        self.backgroundColor = UIColor.timeFrameSegmentBackgroundColor
+    }
+    
+    func config(with buttons: [String]) {
+        self.buttons = buttons
+        
+        self.setupSeparators()
+        self.setupSelectedSegmentView()
+        self.setupStackView()
+        self.setupButtons()
+    }
+    
+    private func setupButtons() {
+        for (i, button) in self.buttons.enumerated() {
+            let font = selectedSegment == i ? labelMediumFont : labelRegularFont
+            
+            let btn = createButton(with: button, font: font, tag: i)
+            btn.addTarget(self, action: #selector(onSegmentButtonTap(_:)), for: .touchUpInside)
+            
+            stackView.addArrangedSubview(btn)
+        }
+    }
+    
+    private func setupStackView() {
+        if stackView.superview != nil {
+            stackView.removeFromSuperview()
+        }
+        
+        stackView = UIStackView()
+        stackView.axis = .horizontal
+        stackView.alignment = .center
+        stackView.distribution = .fillEqually
+        
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        self.addSubview(stackView)
+        
+        stackView.topAnchor.constraint(equalTo: self.topAnchor).isActive = true
```

Comment:
We might use this "view binding" a lot in the future. Move it to `UIView+Bindings` extension to a function like `func bindToSuperview()`